pkg/trace/config/config.go

```diff
@@ -136,6 +136,29 @@ type Tag struct {
 	K, V string
 }
 
+var (
+	isFargate     bool
+	isFargateOnce sync.Once
+)
+
+// IsFargateInstance reports whether we are in a Fargate instance.
+func IsFargateInstance() bool {
+	isFargateOnce.Do(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		go func() {
+			select {
+			case <-ctx.Done():
+				if err := ctx.Err(); err != nil && err != context.Canceled {
```

Comment:
Could we just check `ctx.Err()` after `fargate.IsFargateInstance()` returns rather than spawning a goroutine? I think it would be the same as this, but shorter.

I think either way there's a *minor* chance of a race which could cause the error message to be written even if `isFargate` is set to true. Not sure how much we should be concerned about that.

Reply:
> Could we just check `ctx.Err()` after `fargate.IsFargateInstance()` returns rather than spawning a goroutine? I think it would be the same as this, but shorter. (let me know if I missed something)

It seems so! I tried to follow the example with waiting on the `Done` channel but seems it wasn't necessary. It's much simpler now.

> I think either way there's a *minor* chance of a race which could cause the error message to be written even if `isFargate` is set to true. Not sure how much we should be concerned about that.

That's true, that I did notice. But what's the actual race? `fargate.IsFargateInstance` could finish in 1.999999999s and then the context expires just before the `cancel` call, then technically the call succeeded but we'll get an error. I think I'm ok with that. What we're doing here is we're basically saying that the timeout is in reality not exactly 2s because of the possibility of the race, which is fine because ultimately this is a local HTTP call and 2 seconds is overkill anyway.

Updated.

Reply:
Yes, that's the race I was thinking about. And I agree, the probability is so small and the consequences so minor that I don't think it's worth adding the machinery to get it exact. This looks good to me. :)